model/ftn/gkeModule.f90

```diff
@@ -0,0 +1,1908 @@
+    real, parameter            :: qr_fpow= -5.                ! E(f) tail power law
+!
+    character(len=50), parameter                              &
+                               :: qs_ver  = 'gkev0'           ! version number/str
+    integer, parameter         :: qi_lrb  = 4                 ! 4 bytes
+    real, parameter            :: qr_eps  = epsilon(100.0)    ! Smallest positive
+                                                              ! value supported by the
+                                                              ! compiler (e.g., gfortran
+                                                              ! → 1.19E-7)
+
+    real, parameter            :: qr_grav = 9.806             ! Gravational acc (m/s^2)
+    real, parameter            :: qr_pi   = 3.141592653589793 ! π
+    real, parameter            :: qr_tpi  = 2 * qr_pi         ! π * 2
+    real, parameter            :: qr_dmax = 3000.0            ! Maximum allowed water
+    complex, parameter         :: qc_iu   = (0.0, 1.0)        ! complex unit `i`
+!
+    character(len=100)         :: qs_cfg                      ! File name for quartet/kernel
+!
+    real, allocatable, save    :: qr_kx(:), qr_ky(:),   &     ! kx, ky (2D grid → 1D vector)
+                                  qr_dk(:), qr_om(:)          ! Δ\vec{k}, ω,
+!
+    integer(kind=8)            :: qi_nnz                      ! # of quartets
+    integer                    :: qi_nrsm                     ! # of rows of SMat
+    integer, allocatable, save :: qi_NN(:), qi_PP(:),   &     ! Index for Quartets
+                                  qi_QQ(:), qi_RR(:)
+    real, allocatable, save    :: qr_k4x(:), qr_k4y(:), &     ! kx, ky, ω for 4th wave
+                                  qr_om4(:)
+    real, allocatable, save    :: qr_dom(:),            &     ! Δω
+                                  qr_TKern(:),          &     ! Kernel `T`
+                                  qr_TKurt(:)                 ! Kurtosis `T`
+    integer, allocatable, save :: qi_icCos(:),          &     ! col index of CooCsr
+                                  qi_irCsr(:)                 ! row begining index of
+                                                              ! Csr sparse matrix
+    real, allocatable, save    :: qr_sumQR(:),          &     ! Σ over Q, R
+                                  qr_sumNP(:, :)              ! Σ over P
+!
+    integer, allocatable, save :: qi_bind(:, :)               ! Bilinear interp. (index and
+    real, allocatable, save    :: qr_bwgh(:, :)               ! weight)
+    real, allocatable, save    :: qr_bdry(:)                  ! Boundary weight
+    real, allocatable, save    :: qr_wn1(:)                   ! wavenumber k(nk)
+!/
+!/ ------------------------------------------------------------------- /
+    contains
+!/ ------------------------------------------------------------------- /
+!/ [Part 1]
+!/
+    function QFunc(kx, ky)
+!/
+!/    ??-???-???? : Origination.                        ( O. Gramstad )
```

Comment:
add a date here (and in similar parts below)

Reply:
Done. The approximate date has been added.